core/backend/src/SqliteChangesetReader.ts

```diff
@@ -132,12 +132,13 @@ export class SqliteChangesetReader implements IDisposable {
   /**
    * Open local changes in iModel.
    * @param args iModel and other options.
+   * @param args.db must be of type IModelDb
    * @returns SqliteChangesetReader instance
    */
-  public static openLocalChanges(args: { iModel: IModelJsNative.DgnDb, includeInMemoryChanges?: true } & SqliteChangesetReaderArgs): SqliteChangesetReader {
+  public static openLocalChanges(args: Omit<SqliteChangesetReaderArgs, "db"> & { db: IModelDb, includeInMemoryChanges?: true }): SqliteChangesetReader {
```

Comment:
does the omit still make sense now? since you made db required in the SqliteChangesetReaderArgs?

Reply:
The Omit is just to change the type of `db` when passing SqliteChangesetReaderArgs into this function. The `db` argument is added back in `db: IModelDb`. It looks messy in the function definition, but it just resolves to `SqliteChangesetReaderArgs.db` being of type `IModelDb` here.

It has to be of type `IModelDb` because only that works here. The Omit allows me to override the default value of `db` being `IModelDb | ECDb`.

Reply:
All youre doing here is narrowing the type from `AnyDB` (`IModelDb | ECDb`) to `IModelDB`. Where do we use `ECDb` in here now?

Reply:
Yes. Much simpler way of putting it.

`openGroup()` allows for either database type. And then anywhere else that wants to use a `SqliteChangesetReader` can allow for either Db type to be passed in. In particular, `ChangesetECAdaptor` uses a `SqliteChangesetReader` for reading changesets.